cmd/elasticsearch/cmd/precompute/precompute.go

```diff
@@ -0,0 +1,171 @@
+	for i := 0;; i++ {
+		esResponse, err := queryAndRecomputeAndStore(esClient, path, payload)
+		if err != nil {
+			return err
+		}
+		processedItems += len(esResponse.Hits.Results)
+		logger.Log.Info(fmt.Sprintf("%d%% processed (%d/%d)", processedItems * 100 / esResponse.Hits.Total.Value, processedItems, esResponse.Hits.Total.Value))
+
+		// once we hit the first page with less than pageSize results, there will be no further page -> done
+		if len(esResponse.Hits.Results) < pageSize {
+			break
+		}
+
+		// on to the next page
+		scrollID := esResponse.ScrollID
+		path, payload = BuildScrollQueryNextPage(scrollID)
+	}
+
+	return nil
+}
+
+// queryAndRecomputeAndStore queries the testmachinery index for tm data, recomputes the pre-calculated fields and if changed, updates the respective data
+func queryAndRecomputeAndStore(esClient elasticsearch.Client, path string, payload io.Reader) (QueryResponse, error) {
+	ctx := context.Background()
+	defer ctx.Done()
+
+	bytes, err := esClient.RequestWithCtx(ctx, http.MethodGet, path, payload)
+	if err != nil {
+		return QueryResponse{}, err
+	}
+
+	var esResponse QueryResponse
+	if err := json.Unmarshal(bytes, &esResponse); err != nil {
+		return QueryResponse{}, err
+	}
+	logger.Log.V(6).Info("Got response from ES", "esResponse", esResponse)
+
+	var modifiedItems []Result
+	for _, result := range esResponse.Hits.Results {
+		var k8sVersion, clusterDomain string
+		if result.StepSummary.Metadata != nil {
+			k8sVersion = result.StepSummary.Metadata.Metadata.KubernetesVersion
+			if k8sVersion == "" && result.StepSummary.Metadata.Annotations != nil {
+				k8sVersion = result.StepSummary.Metadata.Annotations["metadata.testmachinery.gardener.cloud/k8sVersion"]
+			}
+			if k8sVersion == "" && result.StepSummary.Metadata.Annotations != nil {
+				k8sVersion = result.StepSummary.Metadata.Annotations["testrunner.testmachinery.sapcloud.io/k8sVersion"]
+			}
+		}
+		oldPreComputed := result.StepSummary.PreComputed
+		if oldPreComputed != nil {
+			clusterDomain = oldPreComputed.ClusterDomain
+		}
+		newPreComputed := collector.PreComputeTeststepFields(result.StepSummary.Phase, k8sVersion, clusterDomain)
+
+		if reflect.DeepEqual(oldPreComputed, newPreComputed) {
+			logger.Log.V(6).Info("old and new precomputed are equal -> nothing to do")
+		} else {
+			logger.Log.Info("preComputed field changed", "oldPreComputed", oldPreComputed, "newPreComputed", newPreComputed)
+			result.StepSummary.PreComputed = newPreComputed
+			modifiedItems = append(modifiedItems, result)
+		}
+	}
+
+	if len(modifiedItems) == 0 {
+		logger.Log.V(6).Info("no modified data for any result (in this page) -> nothing to update")
+		return esResponse, nil
+	}
+
+	path, payload, err = BuildBulkUpdateQuery(modifiedItems)
+	if err != nil {
+		return QueryResponse{}, err
+	}
+	logger.Log.V(6).Info("going to bulk update", "path", path)
+
+
+	if !updateES {
+		logger.Log.Info("Not modifying elasticsearch data as update flag was not set")
```

Comment:
maybe logging the build path and payload would also help debugging when doing a dry run

Reply:
Good idea, done.